smartsim/experiment.py

```diff
@@ -159,28 +161,63 @@ def __init__(
             exp_path = osp.join(getcwd(), name)
 
         self.exp_path = exp_path
+        """The path under which the experiment operate"""
 
-        self._launcher = launcher.lower()
-
-        if self._launcher == "auto":
-            self._launcher = detect_launcher()
-        if self._launcher == "cobalt":
-            raise SSUnsupportedError("Cobalt launcher is no longer supported.")
+        # TODO: Remove this! The controller is becoming obsolete
+        self._control = Controller(launcher="local")
 
-        if launcher == "dragon":
-            self._set_dragon_server_path()
+        self._active_launchers: set[LauncherProtocol[t.Any]] = set()
+        """The active launchers created, used, and reused by the experiment"""
 
-        self._control = Controller(launcher=self._launcher)
-
-        self.fs_identifiers: t.Set[str] = set()
+        self._fs_identifiers: t.Set[str] = set()
+        """Set of feature store identifiers currently in use by this
+        experiment"""
         self._telemetry_cfg = ExperimentTelemetryConfiguration()
+        """Switch to specify if telemetry data should be produced for this
+        experiment"""
+
+    def start_jobs(
+        self, job: Job, *jobs: Job, dispatcher: Dispatcher = DEFAULT_DISPATCHER
+    ) -> tuple[LaunchedJobID, ...]:
+        """Execute a collection of `Job` instances.
+
+        :param job: The job instance to start
+        :param jobs: A collection of other job instances to start
+        :param dispatcher: The dispatcher that should be used to determine how
+            to start a job based on its settings. If not specified it will
+            default to a dispatcher pre-configured by SmartSim.
+        :returns: A sequence of ids with order corresponding to the sequence of
+            jobs that can be used to query or alter the status of that
+            particular execution of the job.
+        """
 
-    def _set_dragon_server_path(self) -> None:
-        """Set path for dragon server through environment varialbes"""
-        if not "SMARTSIM_DRAGON_SERVER_PATH" in environ:
-            environ["SMARTSIM_DRAGON_SERVER_PATH_EXP"] = osp.join(
-                self.exp_path, CONFIG.dragon_default_subdir
-            )
+        def _start(job: Job) -> LaunchedJobID:
+            args = job.launch_settings.launch_args
+            env = job.launch_settings.env_vars
+            # >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+            # FIXME: Remove this cast after `SmartSimEntity` conforms to
+            #        protocol. For now, live with the "dangerous" type cast
+            # ---------------------------------------------------------------------
+            exe = t.cast("ExecutableProtocol", job.entity)
+            # <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+            dispatch = dispatcher.get_dispatch(args)
```

Comment:
Okay so I am trying to understand this, start_jobs accepts a Dispatcher and defaults to a Dispatcher you created in dispatch.py where the Dispatch init accepts the type of LaunchArgument child class. But you are not passing in the LaunchArgument type when initializing the Dispatcher so how can you run `.get_dispatch` when the `self._dispatch_registry` seems like it would be empty?

Reply:
> But you are not passing in the LaunchArgument type when initializing the Dispatcher so how can you run `.get_dispatch` when the `self._dispatch_registry` seems like it would be empty?

Absolutely correct *BUT* the `dispatch` function in `smartsim.settings.dispatch` is actually a pointer to the `DEFAULT_DISPATHCER.dispatch` method. That way when a user creates a new settings and python loads the definition of the `LaunchArguments` subclass it will automatically be added in the `DEFAULT_DISPATCHER._dispatch_registry`

```py
from smartsim.settings.dispatch import dispatch
from smartsim.settings.arguments.launchArguments import LaunchArguments

@dispatch(...)  # <-- Added into the registry here
                #     before the `__init__` method is ever even called
class MyLaunchArguments(LaunchArguments):
     ...
```